src/server/auth/server/api_server.go

```diff
@@ -1477,70 +867,17 @@ func (a *apiServer) GetAuthToken(ctx context.Context, req *auth.GetAuthTokenRequ
 
 // GetAuthToken implements the protobuf auth.GetAuthToken RPC
 func (a *apiServer) GetAuthTokenInTransaction(txnCtx *txnenv.TransactionContext, req *auth.GetAuthTokenRequest) (resp *auth.GetAuthTokenResponse, retErr error) {
-	if !a.isActive() {
+	if err := a.isActive(); err != nil {
```

Comment:
Just checking my understanding here: it looks like this function is changing to drop support for the "no subject" case (also based on the fact that `TestGetAuthTokenNoSubject` is getting removed). Does this need to validate `req.Subject`? (or does that happen elsewhere? I might not have gotten to it yet).

FWIW I'm actually realizing that a consequence of dropping support for github auth is that a bunch of APIs that used to pretty flexible about subjects might need to do more validation now (i.e. subjects/principles should always have a prefix now. Apologies if that is generally checked already, I just need to revisit the code to remind myself).

Reply:
This is definitely a good point. Yes, this drops support for getting a token for yourself. Long-term I was going to split this into "get a token for a pipeline" and "get a token for a robot", since they have pretty different permissions models IMO.